src/net/udp.rs

```diff
@@ -51,14 +54,30 @@ impl UdpSocket {
         let (addr, mut packet_data) = self.state.pre_process_packet(packet, &self.config)?;
 
         let mut bytes_sent = 0;
-
-        for payload in packet_data.parts() {
-            bytes_sent += self
-                .socket
-                .send_to(&payload, addr)
-                .map_err(|io| NetworkError::from(NetworkErrorKind::IOError { inner: io }))?;
+        if cfg!(feature = "tester") {
```

Comment:
Don't think this does the selective compilation? This will check at runtime?

Reply:
https://doc.rust-lang.org/book/first-edition/conditional-compilation.html

Scroll to bottom please and tell me it doesn't mean what I think it means

Reply:
Okay, so it still compiles the code just makes the branch unreachable whereas with the attribute it doesn't even compile that branch?

Reply:
Yes, I think so? What I am wanting to avoid is having an entirely new module

Reply:
Right, I know this might be a bit messy. What I want to avoid is exposing testing interfaces to the user as well.

but now that I think about it, it might be useful to say, flip a switch in amethyst and simulate poor link quality. 🤔 

Maybe we can make it a trait and have different types.

Reply:
The trait idea had occurred to me, and people have expressed interest in having it available during runtime. Honestly the if check is going to add latency on the order of `ns` is that. For 0.1.0, we could just leave it in-line and use benchmarks to see if it causes problems?

Reply:
Are you sure, if there is a hard coded pass through, like say if `None` is used would LLVM optimize it out? aka remove the unreachable branch?

Reply:
I don't know about LLVM optimizing it out, but a boolean if check is very, very fast.